### PR TITLE
fix(playback-watcher): ignore subtitles

### PR DESCRIPTION
**Pull request by gkatsev**
It's possible that a subtitle track will have segments without any text
in them depending on how the subtitles and captions were created.
Therefor, we should ignore it in the playback-watcher where we're
checking for excessive downloads.

### CODE REVIEW

File: src/playback-watcher.js [hunk 1]
```diff
@@ -219,18 +218,6 @@ export default class PlaybackWatcher {
     this.resetSegmentDownloads_(type);
     this.tech_.trigger({type: 'usage', name: `vhs-${type}-download-exclusion`});
 
-    if (type === 'subtitle') {
```
Comment (gesinger) [inline src/playback-watcher.js:222]: I think by removing this we will still try to blacklist subtitle tracks when we don't have subtitles in the segments. Should we just return early for subtitles tracks, or, in loaderTypes above, remove subtitles?

Comment (gkatsev) [reply src/playback-watcher.js:222]: oh, we'll still try to exclude it but at least it won't be removed. Yeah, exiting early here when it's of `type` `subtitle` sounds good.

Comment (gkatsev) [reply src/playback-watcher.js:222]: Updated to return here but not remove the track.

File: test/playback-watcher.test.js [hunk 2]
```diff
@@ -933,64 +933,64 @@ QUnit.module('PlaybackWatcher download detection', {
 });
 
 loaderTypes.forEach(function(type) {
-  QUnit.test(`detects ${type} appends without buffer changes and excludes`, function(assert) {
-    this.setup();
-    const loader = this.mpc[`${type}SegmentLoader_`];
-    const track = {label: 'foobar', mode: 'showing'};
+  if (type !== 'subtitle') {
```
Comment (brandonocasey) [inline test/playback-watcher.test.js:936]: perhaps we should just return in here if `type === 'subtitle'` would keep the changes down. LGTM otherwise.

Comment (gkatsev) [reply test/playback-watcher.test.js:936]: I tried that but it didn't seem to work.

Comment (gesinger) [reply test/playback-watcher.test.js:936]: That seems strange. It wouldn't execute for other loader types of returning early on subtitles?

File: test/playback-watcher.test.js [hunk 3]
```diff
@@ -933,64 +933,64 @@ QUnit.module('PlaybackWatcher download detection', {
...
+    QUnit.test(`detects ${type} appends without buffer changes and excludes`, function(assert) {
+      this.setup();
+      const loader = this.mpc[`${type}SegmentLoader_`];
+      const track = {label: 'foobar', mode: 'showing'};
 
-    if (type === 'subtitle') {
-      loader.track = () => track;
-      sinon.stub(this.player.tech_.textTracks(), 'removeTrack');
-    }
+      if (type === 'subtitle') {
```
Comment (gesinger) [inline test/playback-watcher.test.js:942]: Should this past be removed because we return early on subtitles?

File: test/playback-watcher.test.js [hunk 4]
```diff
@@ -933,64 +933,64 @@ QUnit.module('PlaybackWatcher download detection', {
...
+      const expectedUsage = {};
 
-    expectedUsage[`vhs-${type}-download-exclusion`] = 1;
+      expectedUsage[`vhs-${type}-download-exclusion`] = 1;
 
-    if (type !== 'subtitle') {
-      expectedUsage['vhs-rendition-blacklisted'] = 1;
-      expectedUsage['hls-rendition-blacklisted'] = 1;
-    }
+      if (type !== 'subtitle') {
```
Comment (gesinger) [inline test/playback-watcher.test.js:962]: This conditional probably can be removed.

File: test/playback-watcher.test.js [hunk 5]
```diff
@@ -933,64 +933,64 @@ QUnit.module('PlaybackWatcher download detection', {
...
+        expectedUsage['vhs-rendition-blacklisted'] = 1;
+        expectedUsage['hls-rendition-blacklisted'] = 1;
+      }
 
-    assert.deepEqual(this.usageEvents, expectedUsage, 'usage as expected');
+      assert.deepEqual(this.usageEvents, expectedUsage, 'usage as expected');
 
-    if (type !== 'subtitle') {
-      const message = 'Playback cannot continue. No available working or supported playlists.';
+      if (type !== 'subtitle') {
```
Comment (gesinger) [inline test/playback-watcher.test.js:969]: This conditional probably can be removed, and the else as well.

File: src/playback-watcher.js [hunk 6]
```diff
@@ -220,14 +219,6 @@ export default class PlaybackWatcher {
     this.tech_.trigger({type: 'usage', name: `vhs-${type}-download-exclusion`});
 
     if (type === 'subtitle') {
```
Comment (gesinger) [inline src/playback-watcher.js:221]: Might be worth having a comment here. Something like `// subtitles aren't guaranteed to be present for every bit of video, therefore, we can't determine whether downloads have stalled based on the subtitle's buffer`

**Review by gesinger — COMMENTED** [review]
Sorry for the late review, just catching up.

**Pull request merged**